### ISSUE TITLE
Backport demo configuration related changes to 2.x

### ISSUE THREAD

**Issue by DarshitChanpura** [opensearch-project/security#3827]
This issue serves as a tracker to backport all admin demo configuration related changes to 2.x. Here are the PR that should be backported:

- [ ] https://github.com/opensearch-project/security/pull/3669
- [ ] https://github.com/opensearch-project/security/pull/3734
- [ ] https://github.com/opensearch-project/security/pull/3777
- [ ] https://github.com/opensearch-project/security/pull/3807

Apart from this, there are individual repositories that have references to install demo configuration script. Updates to these instances are tracked in this issue: https://github.com/opensearch-project/security/issues/3624
- [ ] Verify that the changes to demo script reference across different repositories have been backported to 2.x

**Comment by peternied** [comment]
@derek-ho Can this be closed if all the backports are complete?

**Comment by derek-ho** [comment]
Yes I believe so. I have a draft revert PR open in case we need to revert for downstream. If we need to merge that in, I will re-open this.
